GDAPI/GDAPI.Tests/Application/Editors/IDMigrationTests.cs

```diff
@@ -0,0 +1,295 @@
+                        return false;
+                for (int i = 0; i < 5; i++)
+                {
+                    if (collisionTriggers[i].PrimaryBlockID != i + offsetA)
+                        return false;
+                    if (collisionTriggers[i].SecondaryBlockID != i + offsetB)
+                        return false;
+                }
+                return true;
+            }
+        }
+        #endregion
+
+        #region ID Reallocation
+        private static readonly int[] primaryTargetIDs = { 1, 2, 3, 4, 5 };
+        private static readonly int[] secondaryTargetIDs = { 6, 7, 8, 9, 10 };
+        private static readonly int[] primaryTargetIDsWithIgnored = { 30, 40, 1, 2, 3 };
+        private static readonly int[] secondaryTargetIDsWithIgnored = { 901, 902, 4, 5, 6 };
+
+        [Test]
+        public void GroupIDReallocation()
+        {
+            RunIDReallocationTest(IDMigrationMode.Groups, AssertGroupIDReallocation);
+        }
+        [Test]
+        public void ColorIDReallocation()
+        {
+            RunIDReallocationTest(IDMigrationMode.Colors, AssertColorIDReallocation, AssertExtraColorIDReallication);
+        }
+        [Test]
+        public void ItemIDReallocation()
+        {
+            RunIDReallocationTest(IDMigrationMode.Items, AssertItemIDReallocation);
+        }
+        [Test]
+        public void BlockIDReallocation()
+        {
+            RunIDReallocationTest(IDMigrationMode.Blocks, AssertBlockIDReallocation);
+        }
+
+        private void AssertGroupIDReallocation(int i, int id1, int id2)
+        {
+            Assert.AreEqual(id1, normalBlocks[i].GroupIDs[0]);
+            Assert.AreEqual(id1, moveTriggers[i].TargetGroupID);
+            Assert.AreEqual(id1, instantCountTriggers[i].TargetGroupID);
+            Assert.AreEqual(id1, collisionTriggers[i].TargetGroupID);
+        }
+        private void AssertColorIDReallocation(int i, int id1, int id2)
+        {
+            Assert.AreEqual(id1, normalBlocks[i].Color1ID);
+            Assert.AreEqual(id2, normalBlocks[i].Color2ID);
+            Assert.AreEqual(id1, colorTriggers[i].TargetColorID);
+            Assert.AreEqual(id2, colorTriggers[i].CopiedColorID);
+        }
+        private void AssertItemIDReallocation(int i, int id1, int id2)
+        {
+            Assert.AreEqual(id1, pickupItems[i].TargetItemID);
+            Assert.AreEqual(id1, pickupTriggers[i].TargetItemID);
+            Assert.AreEqual(id1, instantCountTriggers[i].ItemID);
+        }
+        private void AssertBlockIDReallocation(int i, int id1, int id2)
+        {
+            Assert.AreEqual(id1, collisionBlocks[i].BlockID);
+            Assert.AreEqual(id1, collisionTriggers[i].PrimaryBlockID);
+            Assert.AreEqual(id2, collisionTriggers[i].SecondaryBlockID);
+        }
+
+        private void AssertExtraColorIDReallication(List<SourceTargetRange> steps, List<Range> ignoredRanges)
+        {
+
+        }
```

Comment:
```suggestion
        { }
```

Reply:
This shouldn't be empty in the first place, it's a missing portion of the tests.